protocols/kad/src/handler.rs

```diff
@@ -491,8 +488,6 @@ impl Handler {
             }
         }
 
-        let keep_alive = KeepAlive::Until(Instant::now() + idle_timeout);
```

Comment:
Hmm, this isn't a clean refactoring at the moment because we are now removing a piece of functionality here: The `idle_timeout`. Whilst we do want to remove that (see https://github.com/libp2p/rust-libp2p/issues/3844), I'd like us to deprecate this functionality first to ensure users don't run into any surprises.

I think for this PR, can't transition any of the protocols that still use `KeepAlive::Until` so we'll have to revert this bit here.

Reply:
reverted b87b155